File: project/BBWT.Client/src/app/main/runtime-editor/rte-dictionary.ts
export class RteDictionaryItem {
    rteId: string;
    phrase: string;
    type: RteDictionaryItemTypes;
    attrs: RteDictionaryItemAttr[];
}

export enum RteDictionaryItemTypes {
    None = 0,
    Html = 1,
    TypeScript = 2,
    CSharp = 3
}

export class RteDictionaryItemAttr {
    attr: string;
    value: string;
}

File: modules/BBWM.RuntimeEditor/RteDictionary.cs
namespace BBWM.RuntimeEditor;

public enum RteDictionaryItemType { None, Html, TypeScript, CSharp };

public class RteDictionaryItemAttr
{
    public string Attr { get; set; }
    public string Value { get; set; }
}

public class RteDictionaryItem
{
    public string RteId { get; set; }
    public RteDictionaryItemType Type { get; set; }
    public string Phrase { get; set; }
    public RteDictionaryItemAttr[] Attrs { get; set; }
}

public class RteDictionary
{
    public RteDictionaryItem[] Items { get; set; }
}
